Client/Frontend/Browser/TopTabsViewController.swift

```diff
@@ -0,0 +1,714 @@
+        attributes.append(decorationAttributes)
+        
+        return attributes
+    }
+    
+    override func layoutAttributesForDecorationViewOfKind(elementKind: String, atIndexPath indexPath: NSIndexPath) -> UICollectionViewLayoutAttributes? {
+        return super.layoutAttributesForDecorationViewOfKind(elementKind, atIndexPath: indexPath)
+    }
+}
+
+private class TopTabFader: UIView {
+    lazy var hMaskLayer: CAGradientLayer = {
+        let innerColor: CGColorRef = UIColor(white: 1.0, alpha: 1.0).CGColor
+        let outerColor: CGColorRef = UIColor(white: 1.0, alpha: 0.0).CGColor
+        let hMaskLayer = CAGradientLayer()
+        hMaskLayer.colors = [outerColor, innerColor, innerColor, outerColor]
+        hMaskLayer.locations = [0.00, 0.03, 0.97, 1.0]
+        hMaskLayer.startPoint = CGPointMake(0, 0.5)
+        hMaskLayer.endPoint = CGPointMake(1.0, 0.5)
+        hMaskLayer.anchorPoint = CGPointZero
+        return hMaskLayer
+    }()
+    
+    init() {
+        super.init(frame: CGRectZero)
+        layer.mask = hMaskLayer
+    }
+    
+    private override func layoutSubviews() {
+        super.layoutSubviews()
+        hMaskLayer.locations = [0.00, 15/frame.width, 1-15/frame.width, 1.0]
+        hMaskLayer.frame = CGRectMake(0, 0, frame.width, frame.height)
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+}
+
+class TopTabsBackgroundDecorationView : UICollectionReusableView {
+    static let Identifier = "TopTabsBackgroundDecorationViewIdentifier"
+    private lazy var right = SingleCurveView(right: true)
+    private lazy var left = SingleCurveView(right: false)
+    
+    lazy var centerBackground: UIView = {
+        let centerBackground = UIView()
+        centerBackground.backgroundColor = TopTabsUX.TopTabsBackgroundNormalColorInactive
+        return centerBackground
+    }()
+    
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        
+        self.addSubview(right)
+        self.addSubview(left)
+        self.addSubview(centerBackground)
+        
+        right.snp_makeConstraints { make in
+            make.right.equalTo(self)
+            make.top.equalTo(self)
+            make.bottom.equalTo(self)
+            make.width.equalTo(SingleCurveView.CurveWidth)
+        }
+        left.snp_makeConstraints { make in
+            make.left.equalTo(self)
+            make.top.equalTo(self)
+            make.bottom.equalTo(self)
+            make.width.equalTo(SingleCurveView.CurveWidth)
+        }
+        centerBackground.snp_makeConstraints { make in
+            make.left.equalTo(left.snp_right)
+            make.right.equalTo(right.snp_left)
+            make.top.equalTo(self)
+            make.bottom.equalTo(self)
+        }
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    override func layoutSubviews() {
```

Comment:
Nit: Instead of overriding `layoutSubview` to call `setNeedsDisplay`, you can set the `contentMode` property in the init method to `.Redraw` which will do this for you.